### PR TITLE
Add documentation for geometry

### PR DESCRIPTION
**Pull request by ezrayst**
- Add figure for euler angle
- Add documentation for rotation matrix

### CODE REVIEW

**Comment by ezrayst** [comment]
Preview of the page can be seen here: https://ezrayst.github.io/mpm-doc/#/theory/mathematics/geometry

File: _sidebar.md [hunk 1]
```diff
@@ -20,6 +20,8 @@
   * [About](theory/about.md)
   * Continuum mechanics
     * [Strains](theory/continuum-mechanics/strain.md)
+  * Mathematics
```
Comment (kks32) [inline _sidebar.md]: Keep it as `geometry`

File: theory/mathematics/geometry.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Rotation matrix {docsify-ignore}
+
+Any 3 dimensional coordinate system (including conventional Cartesian axes x, y and z) can be rotated to another 3 dimensional coordinate system as long as they maintain the orthogonality among the axes. In linear algebra, this is called change of basis.
```
Comment (kks32) [inline theory/mathematics/geometry.md]: either use `three-dimensional` or 3D, I'd suggest `three-dimensional`

Comment (kks32) [inline theory/mathematics/geometry.md]: called `the` change of basis.

File: theory/mathematics/geometry.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Rotation matrix {docsify-ignore}
+
+Any 3 dimensional coordinate system (including conventional Cartesian axes x, y and z) can be rotated to another 3 dimensional coordinate system as long as they maintain the orthogonality among the axes. In linear algebra, this is called change of basis. 
+
+Thus any properties described by a vector in 3 dimensional axes could be rotated in a new set of axes too with the equation below.
```
Comment (kks32) [inline theory/mathematics/geometry.md]: Thus any `property` described `as` a vector in three-dimensional axes could be rotated `to` a new set of axes`:`

File: theory/mathematics/geometry.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+$$ \hat{\mathbf{u}} = \mathbf{R}^{-1} \mathbf{u} $$
+
+where $\hat{\mathbf{u}}$ is property vector (such as velocity and acceleration) in a rotated coordinate system, $\mathbf{R}$ is rotation matrix, and $\mathbf{u}$ is property vector in original coordinate system.
```
Comment (kks32) [inline theory/mathematics/geometry.md]: `the` property vector (`for example` velocity `or` acceleration)

Comment (kks32) [inline theory/mathematics/geometry.md]: is `the` rotation matrix

Comment (kks32) [inline theory/mathematics/geometry.md]: is `the` property vector in `the` original coordinate system.

File: theory/mathematics/geometry.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+The rotation angle convention is shown in figure below.
```
Comment (kks32) [inline theory/mathematics/geometry.md]: `the` figure below

File: theory/mathematics/geometry.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+$$ \hat{\mathbf{u}} = \mathbf{R}^{-1} \mathbf{u} $$
+
+where $\hat{\mathbf{u}}$ is property vector (such as velocity and acceleration) in a rotated coordinate system, $\mathbf{R}$ is rotation matrix, and $\mathbf{u}$ is property vector in original coordinate system.
+
+The rotation angle convention is shown in figure below. 
+
+![alt text](Eulerangles.png "Euler Angles convention in the code")
+
+The lower case $x, y, z$ is the original coordinate system, while the upper case $X, Y, Z$ is the rotated coordinate system. The angles shown are defined anti-clockwise from the original coordinate system. Note that all angles in the code are in radians, not degrees. Also $\gamma = 0$ in 2 dimensional case and the implementation omits this extra variable.
```
Comment (kks32) [inline theory/mathematics/geometry.md]: `two-dimensional`

File: theory/mathematics/geometry.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+The lower case $x, y, z$ is the original coordinate system, while the upper case $X, Y, Z$ is the rotated coordinate system. The angles shown are defined anti-clockwise from the original coordinate system. Note that all angles in the code are in radians, not degrees. Also $\gamma = 0$ in 2 dimensional case and the implementation omits this extra variable. 
+
+The rotation matrix, $\mathbf{R}$ is defined as follow
+
+$$
+\mathbf{R} = \left[\begin{matrix} \cos{\alpha} \cos{\beta} - \sin{\alpha} \cos{\gamma} \sin{\beta} & - \cos{\alpha} \sin{\beta} - \sin{\alpha} \cos{\gamma} \cos{\beta} & \sin{\gamma} \sin{\alpha} \\\\ \sin{\alpha} \cos{\beta} + \cos{\alpha} \cos{\gamma} \sin{\beta} &  - \sin{\alpha} \sin{\beta} + \cos{\alpha} \cos{\gamma} \cos{\beta} & - \sin{\gamma} \cos{\alpha} \\\\ \sin{\gamma} \sin{\beta} & \sin{\gamma} \cos{\beta} & \cos{\gamma} \end{matrix}\right]
+$$
+
+In 2 dimensional, the rotation matrix is simplified as follow
```
Comment (kks32) [inline theory/mathematics/geometry.md]: the rotation matrix `reduces to:`

File: theory/mathematics/geometry.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+where $\hat{\mathbf{u}}$ is property vector (such as velocity and acceleration) in a rotated coordinate system, $\mathbf{R}$ is rotation matrix, and $\mathbf{u}$ is property vector in original coordinate system.
+
+The rotation angle convention is shown in figure below. 
+
+![alt text](Eulerangles.png "Euler Angles convention in the code")
+
+The lower case $x, y, z$ is the original coordinate system, while the upper case $X, Y, Z$ is the rotated coordinate system. The angles shown are defined anti-clockwise from the original coordinate system. Note that all angles in the code are in radians, not degrees. Also $\gamma = 0$ in 2 dimensional case and the implementation omits this extra variable. 
+
+The rotation matrix, $\mathbf{R}$ is defined as follow
```
Comment (kks32) [inline theory/mathematics/geometry.md]: is `:` remove anything after is

File: theory/mathematics/geometry.md (hunk 5 above)
Comment (kks32) [inline theory/mathematics/geometry.md]: Remove this line

**Review by kks32 — CHANGES_REQUESTED** [review]

**Review by kks32 — APPROVED** [review]

**Pull request merged**